ServerCore/FileManager.cs

```diff
@@ -22,6 +23,37 @@ public class FileManager
         /// <param name="contents">Contents of the file</param>
         /// <returns>Url of the file in blob storage</returns>
         public static async Task<Uri> UploadBlobAsync(string fileName, int eventId, Stream contents)
+        {
+            CloudBlockBlob blob = await CreateNewBlob(fileName, eventId);
+
+            await blob.UploadFromStreamAsync(contents);
+            return blob.Uri;
+        }
+
+        /// <summary>
+        /// Copies a file in blob storage
+        /// </summary>
+        /// <param name="fileName">Name of the file to upload. The caller should ensure it is safe to use in a URL</param>
+        /// <param name="eventId">Event the file will be used in</param>
+        /// <param name="sourceUri">Uri of the blob to be copied</param>
+        /// <returns>Url of the file in blob storage</returns>
+        public static async Task<Uri> CloneBlobAsync(string fileName, int eventId, Uri sourceUri)
+        {
+            CloudBlockBlob blobSource = new CloudBlockBlob(sourceUri, StorageAccount.Credentials);
+
+            CloudBlockBlob blob = await CreateNewBlob(fileName, eventId);
+            await blob.StartCopyAsync(blobSource);
+
+            while (blob.CopyState.Status == CopyStatus.Pending)
+            {
+                await Task.Delay(1000);
```

Comment:
I'm worried the sleeping could cause page timeouts for the import page, which could leave us in a torn state. Do we actually need to wait for it to finish copying?

Reply:
Removed, good justification thanks